presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergBasic.java

```diff
@@ -30,30 +29,177 @@
 public class TestIcebergBasic
         extends ProductTest
 {
+    // TODO: Spark SQL doesn't yet support decimal.  When it does add it to the test.
+    // TODO: Spark SQL only stores TIMESTAMP WITH TIME ZONE, and Iceberg only supports
+    // TIMESTAMP with no time zone.  The Spark writes/Presto reads test can pass by
+    // stripping off the UTC.  However, I haven't been able to get the
+    // Presto writes/Spark reads test TIMESTAMPs to match.
+
     // see spark-defaults.conf
     private static final String SPARK_CATALOG = "iceberg_test";
     private static final String PRESTO_CATALOG = "iceberg";
-    private static final String TABLE_NAME = "test_iceberg_basic";
-    private static final String SPARK_TABLE_NAME = format("%s.default.%s", SPARK_CATALOG, TABLE_NAME);
-    private static final String PRESTO_TABLE_NAME = format("%s.default.%s", PRESTO_CATALOG, TABLE_NAME);
 
-    @BeforeTestWithContext
-    @AfterTestWithContext
-    public void dropTestTables()
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testPrestoReadingSparkData()
+    {
+        String baseTableName = "test_presto_reading_primitive_types";
+        String sparkTableName = sparkTableName(baseTableName);
+
+        String sparkTableDefinition =
+                "CREATE TABLE %s (" +
+                        "  _string STRING" +
+                        ", _bigint BIGINT" +
+                        ", _integer INTEGER" +
+                        ", _real REAL" +
+                        ", _double DOUBLE" +
+                        ", _boolean BOOLEAN" +
+                        ", _timestamp TIMESTAMP" +
+                        ", _date DATE" +
+                        ") USING ICEBERG";
+        onSpark().executeQuery(format(sparkTableDefinition, sparkTableName));
+
+        String values = "VALUES (" +
+                "'a_string'" +
+                ", 1000000000000000" +
+                ", 1000000000" +
+                ", 10000000.123" +
+                ", 100000000000.123" +
+                ", true" +
+                ", TIMESTAMP '2020-06-28 14:16:00.456'" +
+                ", DATE '1950-06-28'" +
+                ")";
+        String insert = format("INSERT INTO %s %s", sparkTableName, values);
+        onSpark().executeQuery(insert);
+
+        Row row = row(
+                "a_string",
+                1000000000000000L,
+                1000000000,
+                10000000.123F,
+                100000000000.123,
+                true,
+                "2020-06-28 14:16:00.456",
+                "1950-06-28");
```

Comment:
We could use `java.sql` types:
```suggestion
                Timestamp.valueOf("2020-06-28 14:16:00.456"),
                Date.valueOf("1950-06-28"));
```

Reply:
I wish it were true :)